common/logging/src/tracing_logging_layer.rs

```diff
@@ -79,15 +80,28 @@ where
 
         event.record(&mut visitor);
 
-        let mut span_data = Vec::new();
+        let mut span_data = HashMap::new();
         if let Some(scope) = ctx.event_scope(event) {
             for span in scope.from_root() {
```

Comment:
Rather than looping over spans and then removing stuff, could we just take the _last_ (current) span and use its data?

Reply:
I feel like the problem is not only the duplication, but the fact that _all_ instrumented fields are showing up in log events.

Reply:
Chatted to @jimmygchen and we agreed this is the best approach.

We can also get rid of `from_root`, and just take the first span from `scope` (just `scope.next()` should work).